### PR TITLE
Handle leadership change and shutdown placement gracefully

### PR DESCRIPTION
**Pull request by youngbupark**
# Description

This PR is to handle leadership establishment and revoke scenario properly and shutdown placement gracefully to reduce the impact to runtime and the other placement nodes.

1. Add `--cluster-bootstrap` and `--raft-logstore-path`
2. Handle leadership establishment safely
3. Shutdown gRPC and Raft server gracefully
4. log out all Raft logs as debug level

Run three placement nodes:
```
placement --port 50005 --id node0 --initial-cluster node0=127.0.0.1:15050,node1=127.0.0.1:15051,node2=127.0.0.1:15052 --raft-logstore-path log-node0 

placement --port 50006 --id node1 --initial-cluster node0=127.0.0.1:15050,node1=127.0.0.1:15051,node2=127.0.0.1:15052 --raft-logstore-path log-node1 --cluster-bootstrap=false

placement --port 50007 --id node2 --initial-cluster node0=127.0.0.1:15050,node1=127.0.0.1:15051,node2=127.0.0.1:15052 --raft-logstore-path log-node2 --cluster-bootstrap=false
```

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #2377 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by yaron2 — APPROVED** [review]

File: cmd/placement/config.go [hunk 1]
```diff
@@ -59,7 +61,9 @@ func newConfig() *config {
 
 	flag.StringVar(&cfg.raftID, "id", cfg.raftID, "Placement server ID.")
 	flag.StringVar(&cfg.raftPeerString, "initial-cluster", cfg.raftPeerString, "raft cluster peers")
+	flag.BoolVar(&cfg.raftBootStrap, "cluster-bootstrap", cfg.raftBootStrap, "raft cluster bootstrap required")
 	flag.BoolVar(&cfg.raftInMemEnabled, "inmem-store-enabled", cfg.raftInMemEnabled, "Enable in-memory log and snapshot store")
```
Comment (tcnghia) [inline cmd/placement/config.go:65]:
```suggestion
	flag.BoolVar(&cfg.raftInMemEnabled, "inmem-store-enabled", cfg.raftInMemEnabled, "Enable in-memory log and snapshot store. Can only work if --raftLogStorePath is unset.")
```

File: pkg/placement/membership.go [hunk 2]
```diff
@@ -15,32 +16,139 @@ import (
...
+func (p *Service) MonitorLeadership() {
+	var weAreLeaderCh chan struct{}
+	var leaderLoop sync.WaitGroup
+
+	leaderCh := p.raftNode.Raft().LeaderCh()
+
+	for {
+		select {
+		case isLeader := <-leaderCh:
+			switch {
```
Comment (tcnghia) [inline pkg/placement/membership.go:37]: Will we anticipate more cases in this switch? Otherwise, change to `if-else`?

Comment (youngbupark) [reply pkg/placement/membership.go:37]: yes will do that.

**Pull request merged**